packages/ogmios/test/Provider/TxSubmitProvider/OgmiosTxSubmitProvider.test.ts

```diff
@@ -101,6 +116,80 @@ describe('OgmiosTxSubmitProvider', () => {
         CardanoNodeErrors.TxSubmissionErrors.EraMismatchError
       );
     });
+
+    it('throws an error if context has handles, and no handleProvider is passed', async () => {
+      mockServer = createMockOgmiosServer({
+        submitTx: { response: { failWith: { type: 'eraMismatch' }, success: false } }
+      });
+      await listenPromise(mockServer, connection.port);
+      provider = new OgmiosTxSubmitProvider(connection, { logger });
+      await provider.initialize();
+      await provider.start();
+
+      await expect(
+        provider.submitTx({
+          context: {
+            handles: [handleResolution]
+          },
+          signedTransaction: emptyUintArrayAsHexString
+        })
+      ).rejects.toThrowError(/no handleprovider passed/i);
+    });
+
+    it('does not throw an error if handles resolve to same addresses as in context', async () => {
+      mockServer = createMockOgmiosServer({
+        submitTx: { response: { failWith: { type: 'eraMismatch' }, success: false } }
+      });
+      await listenPromise(mockServer, connection.port);
+      provider = new OgmiosTxSubmitProvider(connection, { logger });
+      await provider.initialize();
+      await provider.start();
+
+      await expect(
+        provider.submitTx({
+          context: {
+            handles: [handleResolution]
+          },
+          signedTransaction: emptyUintArrayAsHexString
+        })
+      ).not.toThrowError(/no handleprovider passed/i);
```

Comment:
Would be best to loosen this assertion to prevent potential false positives
```suggestion
      ).not.toThrow();
```

Reply:
Thanks, that'll do it. Just FIY (no change required): `toThrow` will also accept non-errors thrown, while `toThrowError` expects `instanceof Error`